core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1182,23 +1207,27 @@ public boolean isCompatible(String subject,
       throw new InvalidSchemaException("Previous schema not provided");
     }
 
-    CompatibilityLevel compatibility = getCompatibilityLevelInScope(subject);
-    if (compatibility == CompatibilityLevel.NONE) {
-      // optimization to avoid parsing schemas
-      return true;
-    }
-
     List<ParsedSchema> prevParsedSchemas = new ArrayList<>(previousSchemas.size());
     for (Schema previousSchema : previousSchemas) {
       ParsedSchema prevParsedSchema = parseSchema(previousSchema);
       prevParsedSchemas.add(prevParsedSchema);
     }
 
-    ParsedSchema parsedSchema = parseSchema(newSchema);
-    boolean isCompatible = parsedSchema.isCompatible(compatibility, prevParsedSchemas);
-    // Allow schema providers to modify the schema during compatibility checks
-    newSchema.setSchema(parsedSchema.canonicalString());
-    return isCompatible;
+    return isCompatibleWithPrevious(subject, parseSchema(newSchema), prevParsedSchemas);
+  }
+
+  private boolean isCompatibleWithPrevious(String subject,
+                                           ParsedSchema parsedSchema,
+                                           List<ParsedSchema> previousSchemas)
+      throws SchemaRegistryException {
+
+    CompatibilityLevel compatibility = getCompatibilityLevelInScope(subject);
+    if (compatibility == CompatibilityLevel.NONE) {
+      // optimization to avoid parsing schemas
+      return true;
```

Comment:
Is this optimization still happening ? As it is, we are parsing the previous schemas.

Reply:
Yes, this optimization is moved to another method.

Reply:
Ok, I see what you mean.  Yes, it is no longer happening.  I'll remove.

Reply:
Good catch :)